cloud/amazon/rds_v2.py

```diff
@@ -0,0 +1,1142 @@
+    if module.params.get('port'):
+        params.pop('Port')
+        params['DBPortNumber'] = module.params.get('port')
+
+    # Get current instance so we can see if anything was actually modified and set changed accordingly
+    before_modify_instance = get_db_instance(client, module)
+
+    try:
+        response = client.modify_db_instance(**params)
+    except botocore.exceptions.ClientError as e:
+        e = get_exception()
+        module.fail_json(msg=str(e))
+
+    if params.get('apply_immediately'):
+        if NewDBInstanceIdentifier:
+            # Wait until the new instance name is valid
+            new_instance = None
+            while not new_instance:
+                new_instance = get_db_instance(client, module, NewDBInstanceIdentifier)
+                time.sleep(5)
+
+    # Lookup instance again to see if anything was modified
+    after_modify_instance = get_db_instance(client, module)
+    if cmp(before_modify_instance['DBInstances'], after_modify_instance['DBInstances']) != 0:
+        changed = True
+    else:
+        changed = False
+
+    if module.params.get('wait'):
+        if changed:
+            # Wait for status modifying, then wait for status available.
+            # Note: aurora doesn't transition modifying state
+            if before_modify_instance['DBInstances'][0]['Engine'] == 'aurora':
+                status_achieved, response = wait_for_status(client, module, 'available')
+            else:
+                status_achieved, response = wait_for_status(client, module, 'modifying')
+                status_achieved, response = wait_for_status(client, module, 'available')
+
+            if not status_achieved:
+                module.fail_json(msg='Error modifying RDS instance  - please check the AWS console')
+        else:
+            response = get_db_instance(client, module)
+
+    result = response
+    return changed, result
+
+
+def snapshot_db_instance(client, module):
+    changed = False
+    required_vars = ['db_snapshot_identifier','db_instance_identifier']
+    valid_vars = ['tags']
+    params = validate_parameters(required_vars, valid_vars, module)
+
+    db_snapshot = get_db_snapshot(client, module)
+    if db_snapshot['DBSnapshots']:
+        response = db_snapshot
+        changed=False
+    else:
+        try:
+            response = client.create_db_snapshot(**params)
+            changed = True
+        except botocore.exceptions.ClientError as e:
+            e = get_exception()
+            module.fail_json(msg=str(e))
+
+    if module.params.get('wait'):
+        status_achieved, response = wait_for_status(client, module, 'available')
+        if not status_achieved:
+            module.fail_json(msg='Error waiting for RDS snapshot creation - please check the AWS console')
+
+    result = response
+    return changed, result
+
+
+def delete_db_instance_or_snapshot(client, module):
```

Comment:
2 reasons, so that I could stick with command=delete instead of having to use create_instance, create_snapshot etc. I was also trying to keep this as similar to the original module as possible.